src/hardware/serialport/softmodem.cpp

```diff
@@ -29,6 +29,75 @@
 #include "softmodem.h"
 #include "misc_util.h"
 
+static const char phoneValidChars[] = "01234567890*=,;#+>";
+
+class CPhonebookEntry {
+public:
+	CPhonebookEntry(const char *_phone, const char *_address) {
+		safe_strncpy(phone, _phone, sizeof(phone));
+		safe_strncpy(address, _address, sizeof(phone));
+	}
```

Comment:
Suggest using `std::string`'s for both members and the inputs, and simply initializing them straight away, as it reads more naturally.

Reply:
```
	CPhonebookEntry(const std::string& _phone, const std::string& _address)
		: phone(_phone),
		  address(_address) { }
```